### PR TITLE
CAR file pom does not update when an artifact is renamed or deleted #2617

### PR DESCRIPTION
**Pull request by shyabithd**
## Purpose

In current ESB configuration tool, when a ESB configure project is created and upon updating an artifact, composite application project also needs to be updated. This fix address with issue.

Resolves https://github.com/wso2/product-ei/issues/2617

## Goals

Update CAR file pom.xml when the artifact is renamed or deleted.

## Approach

Registered a listener for workspace of the project. This listener receives all event application publishes and hence filtered only the file remove event. When the remove action is called re-trigger the refresh tree method which refreshes the GUI. 

## User stories

GIT issue: CAR file pom does not update when an artifact is renamed or deleted #2617

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: plugins/org.wso2.developerstudio.eclipse.distribution.project/src/org/wso2/developerstudio/eclipse/distribution/project/editor/DistProjectEditorPage.java [hunk 1]
```diff
@@ -1062,4 +1075,31 @@ private void transformProperties(Properties properties) {
...
+	@Override
+	public boolean visit(IResourceDelta delta) throws CoreException {
+		
+		switch (delta.getKind()) {
+	        case IResourceDelta.REMOVED :
+	        	try {
+	         		refreshForm(); 
+	      		} catch (Exception e) {
+	    			// TODO Auto-generated catch block
+	    			System.out.println("changed execption");
```
Comment (Nirothipan) [inline plugins/org.wso2.developerstudio.eclipse.distribution.project/src/org/wso2/developerstudio/eclipse/distribution/project/editor/DistProjectEditorPage.java]: Shall we remove this sys out

**Comment by Nirothipan** [comment]
Does this have a corresponding git issue. If so can you please link it here

File: plugins/org.wso2.developerstudio.eclipse.distribution.project/src/org/wso2/developerstudio/eclipse/distribution/project/editor/DistProjectEditorPage.java (hunk 1 above)
Comment (shyabithd) [reply plugins/org.wso2.developerstudio.eclipse.distribution.project/src/org/wso2/developerstudio/eclipse/distribution/project/editor/DistProjectEditorPage.java]: done

**Comment by shyabithd** [comment]
Nirothipan - Added a GIT issue

File: plugins/org.wso2.developerstudio.eclipse.distribution.project/src/org/wso2/developerstudio/eclipse/distribution/project/editor/DistProjectEditorPage.java [hunk 2]
```diff
@@ -905,11 +909,20 @@ public void run() {
...
 	public void refreshForm() throws Exception {
 		initContent();
-		createTreeContent();
-		txtVersion.setText(getVersion());
-		txtArtifactId.setText(getArtifactId());
-		txtDescription.setText(getDescription());
-		txtGroupId.setText(getGroupId());
+		new Thread(new Runnable() {
+			public void run() {
+		    	try { Thread.sleep(1000); } catch (Exception e) { }
```
[the remark this answers is not in the document]
Comment (shyabithd) [reply plugins/org.wso2.developerstudio.eclipse.distribution.project/src/org/wso2/developerstudio/eclipse/distribution/project/editor/DistProjectEditorPage.java]: formatted and remove thread sleep.

File: plugins/org.wso2.developerstudio.eclipse.distribution.project/src/org/wso2/developerstudio/eclipse/distribution/project/editor/DistProjectEditorPage.java [hunk 3]
```diff
@@ -903,14 +907,26 @@ public void run() {
...
 
-	public void refreshForm() throws Exception {
-		initContent();
-		createTreeContent();
-		txtVersion.setText(getVersion());
-		txtArtifactId.setText(getArtifactId());
-		txtDescription.setText(getDescription());
-		txtGroupId.setText(getGroupId());
-	}
+	/* If the GUI elements are accessed from a thread other than the thread that has created the element
```
Comment (abeykoon) [inline plugins/org.wso2.developerstudio.eclipse.distribution.project/src/org/wso2/developerstudio/eclipse/distribution/project/editor/DistProjectEditorPage.java]: Shall we make this a proper method comment?

File: plugins/org.wso2.developerstudio.eclipse.distribution.project/src/org/wso2/developerstudio/eclipse/distribution/project/editor/DistProjectEditorPage.java [hunk 4]
```diff
@@ -1062,4 +1078,31 @@ private void transformProperties(Properties properties) {
...
+            } catch (Exception e) {
+                log.error("refresh opration failed", e);
+            }
+            break;
+        }
+        return true;
+    }
+
+    @Override
+    public void resourceChanged(IResourceChangeEvent event) {
```
Comment (abeykoon) [inline plugins/org.wso2.developerstudio.eclipse.distribution.project/src/org/wso2/developerstudio/eclipse/distribution/project/editor/DistProjectEditorPage.java]: Method comments are missing

**Pull request merged**